inventory-generation/notifications/main.yaml

```diff
@@ -0,0 +1,45 @@
+---
+- name: Generate email notifications to be consumed by Ansible Tower scheduled jobs
+  hosts: local
+  gather_facts: false
+  
+  tasks:
+    - name: "Fail If Working Directory Is Not Provided"
+      fail:
+        msg: "directory var needs to be provided in order to ensure a safe working directory"
+      when:
+        - directory is undefined or (directory | trim) == ""
+        - email_templates_directory is undefined or (email_templates_directory | trim) == ""
+
+    - name: Read Engagement Data
+      include_vars:
+        file: "{{ directory }}/engagement.json"
+        
+    - name: Set vars based on engagement.json
+      set_fact:
+        start_date: "{{ (start_date | regex_replace('^(.*)T.*$', '\\1') | to_datetime('%Y-%m-%d')).strftime('%d %b %Y') }}"
+        end_date: "{{ (end_date | regex_replace('^(.*)T.*$', '\\1') | to_datetime('%Y-%m-%d')).strftime('%d %b %Y') }}"
+        env_end_date: "{{ (archive_date | regex_replace('^(.*)T.*$', '\\1') | to_datetime('%Y-%m-%d')).strftime('%d %b %Y') }}"
+        ipa_host: "{{ 'ipa.apps.' + (hosting_environments[0].ocp_sub_domain | lower) + '.' + engagement_region | default('dev-1') + '.' + ocp_base_url }}"
+
+    - name: Check for existing email templates
```

Comment:
This task doesn't really do anything - besides of perhaps just printing something useful to the console(?) - should the return values be captured and used for something, or is it mostly for display/debug purposes?

Reply:
These vars are consumed by various email templates similar to the onboarding work. We might consider either moving these items into the jinja2 templates or even having things like the `ipa_host` in the engagement.json if it is more appropriate there?

The dates are really just a pretty print of the timestamps and would make more sense to me in the email templates.

Reply:
Moved into templates, testing now to see how it looks

Reply:
now that you register the output of the `stat` module it makes a lot more sense - looks good.